resources/generate_book.py

```diff
@@ -545,8 +643,24 @@ def make_book(rm: ResourceManager, i18n: I18n, local_instance: bool = False):
             fire_clay_knapping('tfc:fire_clay_knapping/brick', 'The $(l:mechanics/blast_furnace)Blast Furnace$() only accepts fire bricks as insulation.')
         )),
         entry('quern', 'Quern', 'tfc:quern', pages=(
-            image('tfc:textures/gui/book/wip.png'),
-            # todo: quern
+            crafting('tfc:crafting/quern', text_contents='The $(thing)quern$() is a device for grinding items into powder. The base of the quern can be crafted with 3 $(item)smooth stone$() and 3 of any other $(item)stone$() blocks.'),
```

Comment:
More `$(item)` that should be `$(thing)`, but also esp. for item names, they should be capitalized (as they are in the localization of the item name itself). The numbers written out is one of my writing habits.

```suggestion
            crafting('tfc:crafting/quern', text_contents='The $(thing)quern$() is a device for grinding items into powder. The base of the quern can be crafted with three $(thing)Smooth Stone$() and three of any other $(thing)Stone$() blocks.'),
```